api/models.py

```diff
@@ -206,3 +206,45 @@ async def wait_for_node(cls, timeout):
         if timeout > current_time:
             time_delta = timeout - current_time
             await asyncio.sleep(time_delta.total_seconds())
+
+
+class RegressionData(BaseModel):
+    """Regression object details model"""""
+
+    class Config:
+        """Configuration attributes for RegressionData"""
+        use_enum_values = True
+
+    revision: Revision = Field(
+        description='Git revision of regression object'
+    )
+    status: StatusValues = Field(
+        description='Regression object status'
+    )
+
+
+class Regression(Node):
+    """API model for regression tracking"""
+
+    regression_data: List[RegressionData] = Field(
+        description='Regression details'
+    )
+
+    def __init__(self, **data):
+        if data.get('kind') is None:
+            data['kind'] = 'regression'
+        super().__init__(**data)
+
+    @classmethod
+    def validate_params(cls, params: dict):
+        """Validate regression parameters"""
+        ret, msg = Node.validate_params(params)
+        if not ret:
+            return ret, msg
+
+        status = params.get('regression_data.status')
+        if status and status not in [status.value
+                                     for status in StatusValues]:
+            return False, f"Invalid status value '{status}'"
```

Comment:
I think this is already covered by the Pydantic model, right? At least when data is being posted to the API endpoint, if there's an invalid field then it should return an HTTP error before it reaches this point.

Reply:
Yes, in the case of POST regression or node, Pydantic will automatically raise an error.
We have added this for GET nodes or regressions. In case the user passes invalid status or result attribute to the `/nodes` endpoint.

Reply:
I thought the regression would contain a list of nodes, each of them having their own status already:
```python
regression_data: List[Node]
```
That way we can go through the history of results for a particular "test" (any kind of Node) through different revisions.

Did you have something else in mind to implement this?

Reply:
I had in mind to store a list of only regression specific information for a node. As we want a track of revision and status details when we record a new regression. Other details are stored in the Regression model.